### ISSUE TITLE
Defer fails to modify return values when a function has multiple return values

### ISSUE THREAD

**Issue by jrfondren** [edubart/nelua-lang#233]
### Bug description

I knew that `defer` in Odin was defined to not be able to do this: deferred code sees variables only after they've already been copied for the return, so I tested this in Nelua and saw that it seemed to modify returned variables just fine. This resulted in a strange infinite loop that went away when I manually copied the deferred lines to where they should run.

### Code example

```lua
local function f(n: integer)
  defer n = n + 1 end
  return n
end

local function g(n: integer)
  defer n = n + 1 end
  return n, true
end

print(f(5))  -- 6
print(g(5))  -- 5 true
```

Multiple-return case:
```c
  _mulret_1.r1 = n;
  _mulret_1.r2 = true;
  { /* defer */
    n = (n + 1);
  }
  return _mulret_1;
```

### Expected behavior

For the deferred lines to be invoked as if they'd been copied and pasted prior to all exits from the scope.

### Environment

x86_64 linux
Nelua 0.2.0-dev
Build number: 1588
Git date: 2023-09-16 16:20:44 -0300
Git hash: 596fcca5c77932da8a07c249de59a9dff3099495
Semantic version: 0.2.0-dev.1588+596fcca5
Copyright (C) 2019-2022 Eduardo Bart (https://nelua.io/)

**Comment by stefanos82** [comment]
I don't know whether you have seen it or not, but I have just found it and it's kind of baffling:

```c
int nelua_main(int argc, char** argv) {
  {
    nelua_print_1(tmp_f(5));
    {
      nlmulret_nlint64_nlboolean _tmp1 = tmp_g(5);
      nelua_print_2(_tmp1.r1, _tmp1.r2);
    }
  }
  return 0;
}
```

Should not this code be

```c
int nelua_main(int argc, char** argv) {
  {
    nelua_print_1(tmp_f(5));
  }
    {
      nlmulret_nlint64_nlboolean _tmp1 = tmp_g(5);
      nelua_print_2(_tmp1.r1, _tmp1.r2);
    }
  return 0;
}
```

because `print(f(5))` and `print(g(5))` are executed as two separate `defer`s behind the scenes, correct?

**Comment by jrfondren** [comment]
The braces around `nelua_print_2()` are probably there to quickly free `_tmp` from occupying stack space - it's about multiple value returns from `g()` rather than anything getting deferred.

**Comment by stefanos82** [comment]
OK, I have found the bug; the generated C code for `defer` takes place in wrong line(s):

```c
/* Buggy version */
nlmulret_nlint64_nlboolean tmp_g(int64_t n) {
  nlmulret_nlint64_nlboolean _mulret_1;
  _mulret_1.r1 = n;
  _mulret_1.r2 = true;
  { /* defer */
    n = (n + 1);
  }
  return _mulret_1;
}
```

It should be

```c
/* Fixed version*/
nlmulret_nlint64_nlboolean tmp_g(int64_t n) {
  nlmulret_nlint64_nlboolean _mulret_1;
  { /* defer */
    n = (n + 1);
  }
  _mulret_1.r1 = n;
  _mulret_1.r2 = true;
  return _mulret_1;
}
```

**Comment by edubart** [comment]
This is a serious bug, I try to fix it this weekend. Thanks for reporting!

**Comment by edubart** [comment]
I fixed the misbehavior in https://github.com/edubart/nelua-lang/commit/dc479bf4b1ad1335d9caa6cdbf282cc5979f06c0, by definition `defer` should only execute after evaluating all function statements, including return expressions.
